### ISSUE TITLE
[clang-tidy] misc-unused-using-decls incorrectly reports using as unused when introducing templated oeprator"" to overload

### ISSUE THREAD

**Issue by namniav** [llvm/llvm-project#109737]
https://gcc.godbolt.org/z/ssrjxTGo5

```c++
namespace ns1 {
    template<char...Ds>
        requires (sizeof...(Ds) == 1)
    constexpr int operator""_i() { return 1; }
}

namespace ns2 {

    // warning: using decl 'operator""_i' is unused [misc-unused-using-decls]
    using ns1::operator""_i; // this is required ...

    template<char...Ds>
        requires (sizeof...(Ds) == 2)
    constexpr int operator""_i() { return 2; }

    static_assert(1_i == 1); // for this call.
    static_assert(22_i == 2);
}
```
